core/src/main/java/org/radargun/reporting/Timeline.java

```diff
@@ -17,61 +19,123 @@
 public class Timeline implements Serializable, Comparable<Timeline> {
 
    public final int slaveIndex;
-   private Map<String, List<Event>> events = new HashMap<String, List<Event>>();
-   private Map<String, List<Value>> values = new HashMap<String, List<Value>>();
+   /* Events plotted on all charts as marker events. */
+   private Map<String, List<MarkerEvent>> events = new HashMap<>();
+   /* Values plotted in separate charts */
+   private Map<Category, List<Value>> values = new HashMap<>();
    private long firstTimestamp = Long.MAX_VALUE;
    private long lastTimestamp = Long.MIN_VALUE;
 
    public Timeline(int slaveIndex) {
       this.slaveIndex = slaveIndex;
    }
 
-   public void addEvent(String category, IntervalEvent e) {
-      addEvent(category, (Event) e);
-   }
-
-   public void addEvent(String category, TextEvent e) {
-      addEvent(category, (Event) e);
-   }
-
-   private synchronized void addEvent(String category, Event e) {
-      List<Event> cat = events.get(category);
+   public synchronized void addEvent(String category, MarkerEvent e) {
+      List<MarkerEvent> cat = events.get(category);
       if (cat == null) {
-         cat = new ArrayList<Event>();
+         cat = new ArrayList<>();
          events.put(category, cat);
       }
       cat.add(e);
       updateTimestamps(e);
    }
 
-   public synchronized void addValue(String category, Value e) {
+   public synchronized void addValue(Category category, Value e) {
```

Comment:
Yeah. The Values have two groups of events while MarkerEvents uses just one and so just the identifier in the form of String is required. For Values we need to differentiate between System and Custom events.